lib/mix/lib/mix/task.ex

```diff
@@ -122,11 +122,12 @@ defmodule Mix.Task do
 
   Returns the moduledoc or `nil`.
   """
-  @spec moduledoc(task_module) :: String.t() | nil
+  @spec moduledoc(task_module) :: String.t() | nil | false
```

Comment:
Apparently the spec was wrong, hence the build failure. I Fixed it and updated the spec. If the build is green tomorrow morning I will go ahead and merge this. :)

Reply:
Thanks, I was about to push the same thing. :)